amqp_decimal.c

```diff
@@ -0,0 +1,144 @@
+static const PHP5to7_param_long_type_t AMQP_DECIMAL_N_MAX = UINT32_MAX;
+
+
+/* {{{ proto AMQPDecimal::__construct(int $e, int $n)
+ */
+static PHP_METHOD(amqp_decimal_class, __construct)
+{
+    PHP5to7_param_long_type_t e, n;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ll", &e, &n) == FAILURE) {
+		return;
+	}
+
+	if (e < AMQP_DECIMAL_E_MIN) {
+		zend_throw_exception_ex(amqp_value_exception_class_entry, 0 TSRMLS_CC, "Decimal exponent value must be unsigned.");
+		return;
+	}
+
+	if (e > AMQP_DECIMAL_E_MAX) {
+		zend_throw_exception_ex(amqp_value_exception_class_entry, 0 TSRMLS_CC, "Decimal exponent value must be less than %u.", AMQP_DECIMAL_E_MAX);
+		return;
+	}
+    if (n < AMQP_DECIMAL_N_MIN) {
+        zend_throw_exception_ex(amqp_value_exception_class_entry, 0 TSRMLS_CC, "Decimal significand value must be unsigned.");
+        return;
+    }
+
+    if (n > AMQP_DECIMAL_N_MAX) {
+        zend_throw_exception_ex(amqp_value_exception_class_entry, 0 TSRMLS_CC, "Decimal significand value must be less than %u.", AMQP_DECIMAL_N_MAX);
+        return;
+    }
+
+    zend_update_property_long(this_ce, getThis(), ZEND_STRL("exponent"), e TSRMLS_CC);
+    zend_update_property_long(this_ce, getThis(), ZEND_STRL("significand"), n TSRMLS_CC);
+}
+/* }}} */
+
+/* {{{ proto int AMQPDecimal::getExponent()
+Get exponent */
+static PHP_METHOD(amqp_decimal_class, getExponent)
+{
+	PHP5to7_READ_PROP_RV_PARAM_DECL;
+	PHP_AMQP_NOPARAMS();
+
+	PHP_AMQP_RETURN_THIS_PROP("exponent");
+}
+/* }}} */
+
+/* {{{ proto int AMQPDecimal::getSignificand()
+Get E */
+static PHP_METHOD(amqp_decimal_class, getSignificand)
+{
+    PHP5to7_READ_PROP_RV_PARAM_DECL;
+    PHP_AMQP_NOPARAMS();
+
+    PHP_AMQP_RETURN_THIS_PROP("significand");
+}
+/* }}} */
+
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_amqp_decimal_class_construct, ZEND_SEND_BY_VAL, ZEND_RETURN_VALUE, 2)
+	ZEND_ARG_INFO(0, exponent)
+	ZEND_ARG_INFO(0, significand)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_amqp_decimal_class_getExponent, ZEND_SEND_BY_VAL, ZEND_RETURN_VALUE, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_amqp_decimal_class_getSignificand, ZEND_SEND_BY_VAL, ZEND_RETURN_VALUE, 0)
+ZEND_END_ARG_INFO()
+
+
+zend_function_entry amqp_decimal_class_functions[] = {
+	PHP_ME(amqp_decimal_class, __construct, 	arginfo_amqp_decimal_class_construct,	ZEND_ACC_PUBLIC)
+	PHP_ME(amqp_decimal_class, getExponent, 	arginfo_amqp_decimal_class_getExponent,	ZEND_ACC_PUBLIC)
+	PHP_ME(amqp_decimal_class, getSignificand, 	arginfo_amqp_decimal_class_getSignificand,	ZEND_ACC_PUBLIC)
+
+	{NULL, NULL, NULL}
+};
+
+
+PHP_MINIT_FUNCTION(amqp_decimal)
+{
+	zend_class_entry ce;
+
+	INIT_CLASS_ENTRY(ce, "AMQPDecimal", amqp_decimal_class_functions);
+	this_ce = zend_register_internal_class(&ce TSRMLS_CC);
+	this_ce->ce_flags = this_ce->ce_flags | PHP5to7_ZEND_ACC_FINAL_CLASS;
+
+    zend_declare_class_constant_long(this_ce, ZEND_STRL("EXPONENT_MIN"), AMQP_DECIMAL_E_MIN TSRMLS_CC);
```

Comment:
Could you rename the C variable names as well? I feel that would make it easier to follow the code